Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragmentPresenter.java

```diff
@@ -871,7 +895,99 @@ private void addExtensionTypeSettings(ArrayList<SettingsItem> sl, int wiimoteNum
   {
     switch (extentionType)
     {
-      case 1: // Nunchuk
+      case SettingsActivityPresenter.WIIMOTE_EXT_NONE:
+        final int wiimoteNumberAdjusted;
+
+        if (mSettings.isGameSpecific())
+        {
+          wiimoteNumberAdjusted = wiimoteNumber - 4;
+          mSettings.loadWiimoteProfile(mGameID, wiimoteNumberAdjusted);
+        }
+        else
+        {
+          wiimoteNumberAdjusted = wiimoteNumber - 3;
+        }
+
+        AbstractBooleanSetting wiimoteOrientation = new AbstractBooleanSetting()
+        {
+          Boolean valueSelected;
```

Comment:
I'm assuming this is simply some form of caching? This code is not performance critical, and fetching the value of a setting is relatively fast, so I don't really see the need for this. The code would be simpler if you removed this.

Reply:
No. Initially we want to retrieve the boolean value from the INI for `getBoolean`. Once modified via `setBoolean`, the change is not stored in the INI until saving, so further calls to `getBoolean` wouldn't return the updated value. My workaround was to add a variable to make the up-to-date boolean value accessible to all overrides, and read from the INI again only if the setting is deleted.

Reply:
Unless there's something I'm missing, `setBoolean` saves to the in-memory `IniFile` object and `getBoolean` reads from the in-memory `IniFile` object, so removing the variable should work.

Reply:
I changed `getBoolean` to:
```
@Override
public boolean getBoolean(Settings settings)
{
  return getWiimoteSection(wiimoteNumberAdjusted, true)
          .getBoolean(SettingsFile.KEY_WIIMOTE_ORIENTATION, false);
}
```

When changing the orientation with custom game settings, this value doesn't update after changing the value via `setBoolean`. Exiting to the main screen and then returning to the orientation setting displays the updated value. I'm assuming that `getBoolean` is reading from the INI here given this series of events.

Am I misunderstanding something?

Reply:
In `setBoolean`, you are writing to the key `SettingsFile.KEY_WIIMOTE_ORIENTATION + wiimoteNumberAdjusted` when `settings.isGameSpecific() == true`, but `getBoolean` is always reading from the key `SettingsFile.KEY_WIIMOTE_ORIENTATION`. I'm assuming this has something to do with it?

Reply:
Oh ok but the initial retrieved key needs to be `SettingsFile.KEY_WIIMOTE_ORIENTATION` with custom game settings. `SettingsFile.KEY_WIIMOTE_ORIENTATION + wiimoteNumberAdjusted` is for subsequent calls before saving. I'd need to introduce some way to switch between those two so I used a variable.

Reply:
Oh, I forgot that this is one of those garbage settings that are stored in multiple ways :(

Why does it need to be like this, though? Would it not be possible to load from and save to the same key? (Not that I expect you to necessarily know the answer, since the way it works in master is confusing enough that I don't really understand any part of it myself...)

Reply:
Looking at `SettingsFile.saveCustomGameSettings` was the only way I could understand this questionable system. The key is used to determine save location. Unfortunately I think the real motivations for this lie in the C++ side and I'm less familiar with that part of the code.

I think we can agree that changing the way keys are saved is beyond the scope of this PR. Am I good to leave the variable as-is?

Reply:
Well, I don't have a better suggestion, so I suppose you can leave it like this.